dracut/30ignition/mnt-boot.mount

```diff
@@ -0,0 +1,21 @@
+[Unit]
+Description=Ignition Boot Mount
+ConditionPathExists=/dev/disk/by-label/boot
+
+# Wait for multipathd to start to prevent racing over
+# access to the raw device. This ensures that device symlinks
+# are setup properly.
+Wants=multipathd.service
+After=multipathd.service
+
+# Make sure that the device exists
+Requires=dev-disk-by\x2dlabel-boot.device
+After=dev-disk-by\x2dlabel-boot.device
+
+# Shut down before switch-root
+Conflicts=initrd-switch-root.service
+
+[Mount]
+What=/dev/disk/by-label/boot
+Where=/mnt/boot
```

Comment:
Interesting approach to use `/mnt/boot` instead of `/boot`.  Is the idea that this mount won't conflict with all the other places that are mounting at `/boot` today?

Reply:
The Ignition code uses `/mnt/boot_partition` today.  I change it to `/mnt/boot` since `mnt-boot_partition.mount` was ugly, IMO. Trying not to change the whole schema per in one PR. Although I could be convinced otherwise.

Reply:
I don't have a really strong feeling on this, but you did change the "schema" in this one PR already right?

The argument for having it be `/boot` is symmetry with how we (and really other distributions too AFAIK) mount it in the main OS.

But to be clear not a blocker from my side.

Reply:
The reason I'm bringing this up is the mount path will become an "ABI" that will be hard to change later since presumably we'll start relying on it in fedora-coreos-config and other places too.

Reply:
Sold. I _like_ `/boot` better anyway. Retested and pushed.